### PR TITLE
Fix compress crash

### LINKED ISSUE #2311: Unexpected EOF in qemu builder causes crash in Packer 0.8.0
My template works fine with Packer 0.7.5 but crashes with 0.8.0.

https://github.com/tylert/packer-build/blob/master/debian/jessie/base-64.json

On Mac OS X 10.10.3, "packer validate" causes a crash (no qemu binary there to test further).  On Debian 8.1.0 for amd64, "packer validate" and "packer build -only=qemu" both cause the same crash.

```
Template validation failed. Errors are shown below.

Errors validating build 'qemu'. unexpected EOF


!!!!!!!!!!!!!!!!!!!!!!!!!!! PACKER CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Packer crashed! This is always indicative of a bug within Packer.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Packer[1] so that we can fix this.

[1]: https://github.com/mitchellh/packer/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! PACKER CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

Contents of crash.log:
https://gist.github.com/tylert/8f6c157bf3d09c09cf9e

Please confirm this is not simply a "User Too Stupid Error".

### PR DESCRIPTION
**Pull request by cbednarski**
Fixes #2311

### CODE REVIEW

**Comment by cbednarski** [comment]
This bug was introduced [here][1]. The tests would have caught it but weren't run. :( To resolve that I removed `PACKER_ACC=1` so the tests will run as part of `make test`.

Also moved `vars` to the top so `Config` is closer to its usage in the file.

[1]:https://github.com/mitchellh/packer/commit/5241d8c6d6239f770474251ef85095472efb2471

File: post-processor/compress/post-processor.go [hunk 1]
```diff
@@ -35,21 +47,9 @@ type Config struct {
 }
 
 type PostProcessor struct {
-	config *Config
+	config Config
```
Comment (cbednarski) [inline post-processor/compress/post-processor.go]: Crash fix

**Comment by mitchellh** [comment]
LGTM!

**Comment by florentx** [comment]
thank you!

Now I expect the bug-fix release ;-) ... all my builds are crashing with 0.8

**Comment by florentx** [comment]
add an entry for GH-#2311 to the changelog of 0.8.1?

**Comment by cbednarski** [comment]
@florentx Good catch, thanks.

**Pull request merged**